tensorflow_datasets/image/ham10000.py

```diff
@@ -0,0 +1,108 @@
+                "benign keratosis-like lesions (solar lentigines / seborrheic "
+                "keratoses and lichen-planus like keratoses, bkl), dermatofibroma (df), "
+                "melanoma (mel), melanocytic nevi (nv) and vascular lesions "
+                "(angiomas, angiokeratomas, pyogenic granulomas and hemorrhage, vasc)")
+
+_IMAGE_SHAPE = (450, 600, 3)
+
+_NAMES = ['akiec', 'bcc', 'bkl', 'df', 'mel', 'nv', 'vasc'] 
+
+_CITATION = """\
+@data{DVN/DBW86T_2018,
+author = {Tschandl, Philipp},
+publisher = {Harvard Dataverse},
+title = "{The HAM10000 dataset, a large collection of multi-source dermatoscopic images of common pigmented skin lesions}",
+UNF = {UNF:6:IQTf5Cb+3EzwZ95U5r0hnQ==},
+year = {2018},
+version = {V1},
+doi = {10.7910/DVN/DBW86T},
+url = {https://doi.org/10.7910/DVN/DBW86T}
+}
+"""
+
+class Ham10000(tfds.core.GeneratorBasedBuilder):
+    """HAM10000 Dermatoscopic images"""
+
+    VERSION = tfds.core.Version('1.0.0')
+
+
+    def _info(self):
+        return tfds.core.DatasetInfo(
+            builder=self,
+    
+            description=(_DESCRIPTION),
+            
+            features=tfds.features.FeaturesDict({
+                "image":tfds.features.Image(shape=_IMAGE_SHAPE),
+                "label": tfds.features.ClassLabel(names=_NAMES),
+            }),
+
+            supervised_keys=("image", "label"),
+            
+            urls=[_URL],
+            
+            citation=_CITATION
+        )
+
+    def _split_generators(self, dl_manager):
+        """Function to split the images into training set as no other set is specified"""
+        
+        path = dl_manager.manual_dir
+        return [
+            tfds.core.SplitGenerator(
+                name="train",
+                num_shards=10,
+                gen_kwargs={
+                    "images_dir_path": path,
+                    "labels_dir_path":os.path.join(path, "HAM10000_metadata.csv")
+                },
+            )
+        
+        ]
+
+    def _generate_examples(self, images_dir_path, labels_dir_path):
+        """ Function to extract images and labels"""
+
+        csv_dict = readCsv(labels_dir_path)
+        labels_dict = return_map(csv_dict)
+        data_folder_list = [os.path.join(images_dir_path, 'HAM10000_images_part_1'), os.path.join(images_dir_path, 'HAM10000_images_part_2')]
+        for data_folder in data_folder_list:
+            for image_file in tf.io.gfile.listdir(data_folder):
+
+                image_path = os.path.join(data_folder, image_file)
+                label = labels_dict[image_file.split(".")[0]]
+                 
+                yield {
+                
+                    "label": label,
+                    "image": image_path
+                }
+
+
+def readCsv(labels_dir_path):
+    """Function to read labels.csv file and store in memory"""
+
+    readCSV = csv.DictReader(tf.io.gfile.GFile(labels_dir_path))
```

Comment:
as this function is just one line - I'd suggest just putting this line in 83 directly.